src/objects/Points.js

```diff
@@ -160,6 +160,9 @@ Points.prototype = Object.assign( Object.create( Object3D.prototype ), {
 
 function testPoint( point, index, localThresholdSq, matrixWorld, raycaster, intersects, object ) {
 
+	if ( ! object.material.sizeAttenuation )
+		new Ray( _ray.origin ).lookAt( point ).at( new Vector3().distanceTo( _ray.origin ), point );
```

Comment:
Tip: Avoid instantiating new objects -- particularly in methods that are called in tight loops. Create single instances and reuse them.

Reply:
Hope I understand you.
See my new version [testSizeAttenuation](https://github.com/mrdoob/three.js/pull/20144/commits/eecfc03e68845a695c5e88f4c50a92b216b02f1c)